src/coreclr/src/debug/dbgutil/elfreader.cpp

```diff
@@ -237,7 +237,7 @@ ElfReader::InitializeGnuHashTable()
         TRACE("ERROR: InitializeGnuHashTable invalid BucketCount or SymbolOffset\n");
         return false;
     }
-    m_buckets = new (std::nothrow) int32_t[m_hashTable.BucketCount];
+    m_buckets = (int32_t*)malloc(m_hashTable.BucketCount * sizeof(int32_t));
```

Comment:
Any reason to switch from `new[]` to `malloc()`?

Reply:
The runtime has these overrides that replace new/delete with ones in clrhost_nodependencies.cpp but for the elfreader, the "new" wasn't going through those overrides but the "delete" was causing an assert in the debug build. It was simpler to use malloc/free.